### ISSUE TITLE
`AutoCommandBufferBuilder::begin_render_pass` doesn't clear all cubemap layers

### ISSUE THREAD

**Issue by Tenebryo** [vulkano-rs/vulkano#1454]
*   Version of vulkano: `0.19.0` from crates.io
*   OS: Windows 10 x64
*   GPU (the selected PhysicalDevice): GeForce RTX 2070 SUPER
*   GPU Driver: 460.79
*   See bottom of issue for a minimal example

# Issue

`AutoCommandBufferBuilder::begin_render_pass` only clears the first layer of cubemap attachments. After some digging through the source, it appears this is because `Dimensions::array_layers` (called from `FramebufferBuilder::add`) returns 1 instead of 6 for `Dimensions::Cubemap{size}`. I noticed that if you convert the dimensions to a `ImageDimensions` it does return 6.

With my limited familiarity with the codebase, I see two possible solutions:
* ~~change the behavior of `Dimensions::array_layers` so that it returns 6 for cubemaps [here](https://github.com/vulkano-rs/vulkano/blob/master/vulkano/src/image/mod.rs#L304)~~ I didn't see the `Dimensions::array_layers_with_cube` function, so this is probably not the solution
* convert the `Dimensions` to `ImageDimensions` in `FramebufferBuilder::add` (via `Dimensions::to_image_dimensions`) [here](https://github.com/vulkano-rs/vulkano/blob/master/vulkano/src/framebuffer/framebuffer.rs#L181)

# Minimal Bug Reproduction
```rust
use vulkano::buffer::{BufferUsage, CpuAccessibleBuffer};
use vulkano::image::{Dimensions, StorageImage};
use vulkano::instance::InstanceExtensions;
use vulkano::command_buffer::AutoCommandBufferBuilder;
use vulkano::device::{Device, DeviceExtensions};
use vulkano::framebuffer::Framebuffer;
use vulkano::image::ImageUsage;
use vulkano::instance::{Instance, PhysicalDevice};
use vulkano::format::Format;
use vulkano::sync;
use vulkano::sync::GpuFuture;

use std::sync::Arc;

fn main() {

    let instance = Instance::new(None, &InstanceExtensions::none(), None).unwrap();

    let physical = PhysicalDevice::enumerate(&instance).next().unwrap();

    println!(
        "Using device: {} (type: {:?})",
        physical.name(),
        physical.ty()
    );

    let queue_family = physical
        .queue_families()
        .find(|&q| {
            q.supports_graphics()
        })
        .unwrap();

    let device_ext = DeviceExtensions {
        khr_swapchain: true,
        ..DeviceExtensions::none()
    };
    let (device, mut queues) = Device::new(
        physical,
        physical.supported_features(),
        &device_ext,
        [(queue_family, 0.5)].iter().cloned(),
    )
    .unwrap();

    let queue = queues.next().unwrap();

    let render_pass = Arc::new(
        vulkano::single_pass_renderpass!(
            device.clone(),
            attachments: {
                color: {
                    load: Clear,
                    store: Store,
                    format: Format::R32G32B32A32Sfloat,
                    samples: 1,
                }
            },
            pass: {
                color: [color],
                depth_stencil: {}
            }
        )
        .unwrap(),
    );

    let image_buffer = StorageImage::with_usage(
        device.clone(),
        Dimensions::Cubemap{size : 4},
        Format::R32G32B32A32Sfloat,
        ImageUsage{
            transfer_source          : true,
            transfer_destination     : true,
            sampled                  : false,
            storage                  : false,
            color_attachment         : true,
            depth_stencil_attachment : false,
            transient_attachment     : false,
            input_attachment         : true,
        },
        [queue.family()].iter().cloned()
    ).unwrap();

    let out_buffer = CpuAccessibleBuffer::from_iter(
        device.clone(), 
        BufferUsage::all(), 
        false, [0.0f32; 4*4*4*6].iter().cloned()
    ).unwrap();

    assert_eq!(image_buffer.dimensions().array_layers(), 1);
    assert_eq!(image_buffer.dimensions().to_image_dimensions().array_layers(), 6);

    let framebuffer = Arc::new(
        Framebuffer::start(render_pass.clone())
            .add(image_buffer.clone()).unwrap() // Framebuffer think the image only has 1 layer here
            .build().unwrap()
    );

    let clear_values = vec![[0.0, 0.0, 1.0, 1.0].into()];

    let mut builder = AutoCommandBufferBuilder::primary_one_time_submit(
        device.clone(),
        queue.family(),
    )
    .unwrap();

    builder
        // *** render pass only clears layer 0
        .begin_render_pass(framebuffer.clone(), false, clear_values).unwrap()
        .end_render_pass().unwrap()
        .copy_image_to_buffer(image_buffer.clone(), out_buffer.clone()).unwrap();

    let command_buffer = builder.build().unwrap();

    sync::now(device.clone())
        .then_execute(queue.clone(), command_buffer).unwrap()
        .then_signal_fence_and_flush().unwrap()
        .wait(None).unwrap();

    let data = out_buffer.read().unwrap();

    for (i, p) in data.chunks_exact(16).enumerate() {
        if i % 4 == 0 {
            println!("Layer: {}", i / 4);
        }
        println!("  {:?}", p);
    }
    
}
```

**Comment by Eliah-Lakhin** [comment]
@Tenebryo Thank you for the bug report. Would you like to take a lead in fixing it? I'm fine with any solution you choose. It would be just more preferable if it wouldn't break existing public API(unless absolutely unavoidable).

**Comment by Tenebryo** [comment]
@Eliah-Lakhin I created a pull request ( #1459 ) with my proposed fix. The only change is to use `ImageDimensions` instead of `Dimensions` for framebuffer attachments, which should only affect cube map attachments. I've been using this change in the project I found the bug in.

**Comment by Eliah-Lakhin** [comment]
Fixed in #1459
